File: backend/app/api/v1/endpoints/health.py
"""Health check endpoint."""

from datetime import UTC, datetime

from fastapi import APIRouter
from pydantic import BaseModel

from backend.app.core.config import settings

router = APIRouter()


class HealthResponse(BaseModel):
    """Health check response model."""

    status: str
    version: str
    environment: str
    timestamp: datetime
    projects_dir: str
    backups_dir: str


@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """
    Health check endpoint.

    Returns application status and configuration information.
    """
    return HealthResponse(
        status="healthy",
        version=settings.VERSION,
        environment=settings.ENVIRONMENT,
        timestamp=datetime.now(UTC),
        projects_dir=str(settings.PROJECTS_DIR),
        backups_dir=str(settings.BACKUPS_DIR),
    )

File: frontend/src/api/health.ts
/**
 * Health check API
 */

import { apiClient } from './client'

export interface HealthResponse {
  status: string
  version: string
  environment: string
  timestamp: string
  projects_dir: string
  backups_dir: string
}

export const healthApi = {
  /**
   * Get application health and version information
   */
  getHealth: async (): Promise<HealthResponse> => {
    const response = await apiClient.get<HealthResponse>('/health')
    return response.data
  },
}
